actionpack/test/dispatch/request/query_string_parsing_test.rb

```diff
@@ -93,32 +93,34 @@ def teardown
     )
   end
 
-  test "query string without equal" do
-    assert_parses({ "action" => nil }, "action")
-    assert_parses({ "action" => { "foo" => nil } }, "action[foo]")
-    assert_parses({ "action" => { "foo" => { "bar" => nil } } }, "action[foo][bar]")
-    assert_parses({ "action" => { "foo" => { "bar" => [] } } }, "action[foo][bar][]")
-    assert_parses({ "action" => { "foo" => [] } }, "action[foo][]")
-    assert_parses({ "action" => { "foo" => [{ "bar" => nil }] } }, "action[foo][][bar]")
-  end
-
-  def test_array_parses_without_nil
-    assert_parses({ "action" => ["1"] }, "action[]=1&action[]")
-  end
-
-  test "perform_deep_munge" do
-    old_perform_deep_munge = ActionDispatch::Request::Utils.perform_deep_munge
-    ActionDispatch::Request::Utils.perform_deep_munge = false
-    begin
+  if Rack.release < "3"
```

Comment:
Wouldn't it make sense to add the opposite test when `Rack.release >= 3`?

Reply:
I don't have a strong opinion, but this is consistent with the other checks I added. I slightly prefer `if Rack.release < "3"` rather than `unless Rack.release >= "3"`.

Reply:
I wasn't suggesting to use unless but to have an else clause with a rack 3 version of the test. Because here you are basically removing tests.


Anyway, Matthew's concerns would need to be addressed first.

Reply:
I don't think there is any point repeating tests that already exist in Rack. What do you think?